### PR TITLE
Optimize select

### PR DESCRIPTION
**Pull request by Cashmaney**

### CODE REVIEW

**Review by liorbond — CHANGES_REQUESTED** [review]

File: libtfhe-wrapper/src/api/ffi/api.rs [hunk 1]
```diff
@@ -353,34 +358,63 @@ pub unsafe extern "C" fn cmux(
...
+///
+/// # Returns
+/// A `Result` containing either the encrypted result of the cmux operation or a `RustError`.
+///
+/// # Errors
+/// Returns an error if any of the FHE operations fail.
+fn perform_cmux(uint_type: FheUintType, control_slice: &[u8], if_true_slice: &[u8], if_false_slice: &[u8]) -> Result<Vec<u8>, RustError> {
+
+    // Encrypt a 0 value as a base for creating a mask.
+    let mut mask = trivial_encrypt_safe(0, uint_type)?;
```
[the remark this answers is not in the document]
Comment (Cashmaney) [reply libtfhe-wrapper/src/api/ffi/api.rs:394]: It's inside the rust code - this encrypt takes orders of magnitude less time then everything else

File: internal/api/amd64/lib.go [hunk 2]
```diff
@@ -90,7 +90,9 @@ func Cmux(control []byte, ifTrue []byte, ifFalse []byte, uintType uint8) ([]byte
 		return nil, errorWithMessage(err, errmsg)
 	}
 
-	return copyAndDestroyUnmanagedVector(res), nil
+	result := copyAndDestroyUnmanagedVector(res)
+
+	return result, nil
```
[the remark this answers is not in the document]
Comment (Cashmaney) [reply internal/api/amd64/lib.go:95]: Ah, I'll revert. Did this to do some microbenches

File: libtfhe-wrapper/src/api/ffi/api.rs [hunk 3]
```diff
@@ -353,34 +358,63 @@ pub unsafe extern "C" fn cmux(
...
+    // Subtract the control slice from the mask, effectively creating an encryption of (0 - control).
```
[the remark this answers is not in the document]
Comment (Cashmaney) [reply libtfhe-wrapper/src/api/ffi/api.rs:395]: I put the main logic in the function desc as well, but sure

File: libtfhe-wrapper/src/api/ffi/api.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (liorbond) [reply libtfhe-wrapper/src/api/ffi/api.rs:393]: I know but would it be easy to optimize it? it will probably reduce 3ms but why not :P

**Review by liorbond — APPROVED** [review]

**Review by eshelB — COMMENTED** [review]
Nice!

**Pull request merged**